platform/android/export/export.cpp

```diff
@@ -2488,13 +2507,26 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 
 			_update_custom_build_project();
 
+			if (export_app_bundle == 1) {
+				//stores all the project files inside the Gradle project directory. Also includes all ABIs
+				err = export_project_files(p_preset, rename_and_store_file_in_gradle_project, NULL, ignore_so_file);
+				if (err != OK) {
+					EditorNode::add_io_error("Could not export project files to gradle project\n");
+					return err;
+				}
+			}
```

Comment:
@amanj120 Here you're only writing the project to the asset directory if the export format is `aab`. Shouldn't there be a counter-check logic to prevent the legacy system from attempting to update an apk that doesn't exist?

Does this logic also means that apk generation still uses part of the legacy system to push the project resources in the apk?

Reply:
The way the build system currently works is that selecting "Use Custom Build" and "Export APK" still puts project files into the APK by unzipping the APK and putting the contents in there. If we want to change that behavior so that "Use Custom Build" and "export APK" copies the game's project files to the Gradle project, then I can do that.

Reply:
> Also, what do you mean by counter-check logic? If "Use Custom Build" and "Export APK" are selected, then the current solution generates a template APK, just like it does in 3.2.3, so the legacy system will always have an APK to update.

By `counter-check logic`, I was referring to the case where `Use Custom Build` and `Export AAB` are selected. In that scenario, don't we use additional logic to prevent the code from attempting to treat the generated aab binary as an apk?